### PR TITLE
Fix non-accessed field warning

### PR DESCRIPTION
**Pull request by tomkirbygreen**

### CODE REVIEW

File: src/IO.Ably.Shared/ClientOptions.cs [hunk 1]
```diff
@@ -292,7 +292,12 @@ public bool UseBinaryProtocol
 #else
             get { return false; }
 #endif
+
+#if MSGPACK
             set { _useBinaryProtocol = value; }
+#else
```
Comment (sacOO7) [inline src/IO.Ably.Shared/ClientOptions.cs:298]: Do we need `else` statement here? We can just end with `endif`

Comment (sacOO7) [reply src/IO.Ably.Shared/ClientOptions.cs:298]: Also, I am thinking, we can possibly move L297 to under L19. This is optional though

Comment (tomkirbygreen) [reply src/IO.Ably.Shared/ClientOptions.cs:298]: We need the `#else` since we call the setter in code.

Comment (sacOO7) [reply src/IO.Ably.Shared/ClientOptions.cs:298]: Got it. Maybe we could add conditional statement there too. Just that it looks too redundant. I am approving PR for now, you can decide to change it.

**Review by sacOO7 — APPROVED** [review]

**Pull request merged**